src/modules/navigator/MissionFeasibility/FeasibilityChecker.cpp

```diff
@@ -601,50 +602,45 @@ bool FeasibilityChecker::checkTakeoffLandAvailable()
 }
 
 
-bool FeasibilityChecker::checkDistanceToFirstWaypoint(mission_item_s &mission_item)
+bool FeasibilityChecker::checkHorizontalDistanceToFirstWaypoint(mission_item_s &mission_item)
 {
-	if (_param_mis_dist_1wp <= 0.0f || !_home_lat_lon.isAllFinite()) {
-		/* param not set, check is ok */
-		return true;
-	}
+	if (_param_mis_dist_1wp > FLT_EPSILON &&
+	    (_current_position_lat_lon.isAllFinite()) && !_first_waypoint_found &&
+	    MissionBlock::item_contains_position(mission_item)) {
 
-	if (!_first_waypoint_found && MissionBlock::item_contains_position(mission_item)) {
 		_first_waypoint_found = true;
 
+		/* check distance from current position or Home (whatever is closer) to item */
```

Comment:
```suggestion
		/* check distance from current position to item */
```
No longer checked against home.

But I'd probably just delete this comment as it doesnt add anything

Reply:
Yes, removed it.